Description: specify static member variable PrintThread::cout_mutex inline (C++17)
Origin: upstream
Bug: https://github.com/rdkit/rdkit/issues/5351
Applied-Upstream: https://github.com/rdkit/rdkit/commit/48497c40f884c807f5e7ffd95bc7b44e2a3cf2b1
Last-Update: 2022-08-08
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: rdkit/Code/GraphMol/FileParsers/testMultithreadedMolSupplier.cpp
===================================================================
--- rdkit.orig/Code/GraphMol/FileParsers/testMultithreadedMolSupplier.cpp
+++ rdkit/Code/GraphMol/FileParsers/testMultithreadedMolSupplier.cpp
@@ -30,7 +30,7 @@ using namespace std::chrono;
 // thread safe printing for debugging
 // Usage Example: PrintThread{} << "something";
 struct PrintThread : public std::stringstream {
-  static std::mutex cout_mutex;
+  inline static std::mutex cout_mutex;
   ~PrintThread() override {
     std::lock_guard<std::mutex> l{cout_mutex};
     std::cout << rdbuf();
